versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/CaffeineCacheBackend.java

```diff
@@ -145,6 +179,17 @@ public void put(@Nonnull String repositoryId, @Nonnull Obj obj) {
     putLocal(repositoryId, obj);
   }
 
+  @VisibleForTesting
+  void cachePut(CacheKeyValue key, CacheKeyValue value) {
+    var w = weigher(key, value);
+    if (currentWeight.get() + w < admitWeight) {
+      currentWeight.addAndGet(w);
```

Comment:
I wonder if we could leverage the cache's [Policy.Eviction.weightedSize()](https://github.com/projectnessie/nessie/pull/10640/files#diff-89933869cb93756b3343911334c3f3455d7a608f052be1997cbcc42c24b13972R119) :thinking: 

That method could cause inline eviction, so more work on `put`, but we'd get the benefit of actually evicting stale objects rather that rejecting fresh objects... WDYT?

Reply:
It's already above the "max weight" bound in such cases.